jena-geosparql/src/main/java/org/locationtech/jts/index/strtree/CustomIndexSerdeV2.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.locationtech.jts.index.strtree;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.jena.geosparql.spatial.index.v2.serde.sedona.CustomGeometrySerde;
+import org.locationtech.jts.geom.Envelope;
+
+import com.esotericsoftware.kryo.Kryo;
+import com.esotericsoftware.kryo.io.Input;
+import com.esotericsoftware.kryo.io.Output;
+
+/* This file is an adapted copy of org.locationtech.jts.index.strtree.IndexSerde. */
+/* The change is, that geometry read/write is delegated to a GeometrySerdeAdapter instance. */
```

Comment:
This class is copy&adapt form sedona-common 1.7.0 (latest)- they actually used this package hack to access non-visible stuff. in the long run, jts should fix this design.